ledger-service/http-json/src/main/scala/com/digitalasset/http/DbStartupResult.scala

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+
+import com.daml.http.dbbackend.ContractDao
+import com.daml.logging.{ContextualizedLogger, LoggingContextOf}
+import cats.effect.IO
+import com.daml.http.util.Logging.InstanceUUID
+
+sealed trait DbStartupResult
+object DbStartupResult {
+  case object GracefullyExit extends DbStartupResult
```

Comment:
you really don’t seem to like my suggestion to move the logic for this to the call site :grin: fair enough I don’t care too much